src/pages/extending-chat-widget/chat-widget-adapters/index.mdx

```diff
@@ -0,0 +1,763 @@
+
+3. Register provided event handlers for the corresponding events
+
+4. Notify reactive data interface that widget has been initialized to update related data
+
+### Update
+
+This phase allows updating provided data or callbacks by changing their value or reference. In order to do so, the component internally observes for such changes and translate them to appropriate imperative mutations via JS API method calls.
+
+1. If properties were updated then call coresponding setter methods on JS API
+
+2. If properties that require full reload were updated then reinitialize widget with new properties
+
+3. If event handlers were updated then unregister the previous one and register a new one for the corresponding event
+
+### Unmount
+
+This phase results in complete Chat Widget cleanup but allowing for proper Mount phase once again.
+
+1. Unregister all registered callbacks
+
+2. Call destroy method from JS API
+
+3. Notify reactive data interface that widget has been destroyed to clear related data
+
+## Reactive data interface
+
+After looking into existing JS API events and available getters we can distinguish a set of entities that the Chat Widget exposes to interact with it.
+Each entity contains data that may change during Chat Widget lifetime.
+Entites are properly initialized on widget load and cleared after widget is destroyed.
+All posible updates are being notified for the consumer of those data.
+The consumer have a way to subscirbe or unsubscribe for the updates in any given moment during application lifetime.
+The Reactive data interface efectivly impements the Observer pattern in which a subject maintains a list of its observers and notifies them about any state changes, usually by calling one of their methods.
+
+| **Chat** |          |
+| -------- | -------- |
+| chatId   | `string` |
+| threadId | `string` |
+
+| **Greeting** |          |
+| ------------ | -------- |
+| id           | `number` |
+| uniqueId     | `string` |
+
+| **Widget readiness** |           |
+| -------------------- | --------- |
+| isWidgetReady        | `boolean` |
+
+| **Widget state** |                                        |
+| ---------------- | -------------------------------------- |
+| availability     | `'online' / 'offline'`                 |
+| visibility       | `'maximized' / 'minimized' / 'hidden'` |
+
+| **Customer**     |                                                  |
+| ---------------- | ------------------------------------------------ |
+| name             | `string`                                         |
+| email            | `string`                                         |
+| isReturning      | `boolean`                                        |
+| isReturning      | `boolean`                                        |
+| sessionVariables | `Record<string, string>`                         |
+| status           | `'queued' / 'chatting' / 'browsing' / 'invited'` |
+
+# React adapter
```

Comment:
Maybe we could separate the adapters' descriptions as subsections of the main "Chat Widget Adapters" section? Then, they would have their own `index.mdx` files, making it easier for us to sync them between the docs and readme directly in the package (perhaps automatically in the future). WDYT?

Reply:
@klarzynskik current file structure depends on Next.js file routing and that's why a single article needs to be in a single file as they are not concatenated anymore. All of our docs look like this now. The only way to separate them would be to create for each of them a dedicated article but I would like to avoid that and keep everything close together as single peace of documentation dedicated for Chat widget Adapters topic 🙂

Reply:
> The only way to separate them would be to create for each of them a dedicated article

yeah, I understand the routing here, and having them separated was exactly what I was suggesting 😅 

I've meant something like that, similarly to what we have in docs e.g. for the "App Guides" (https://developers.livechat.com/docs/getting-started/app-guides):

![image](https://user-images.githubusercontent.com/47242672/152352594-565e426c-8b5d-44e3-a3c7-7db21ded72a1.png)

however, I understand your point, it's not that urgent, just thought it will be easier to sync and maintain

Reply:
Yes you are right I have wrongly understood your proposal 🙇‍♂️ I thought you want to "code split" the documentation as it was working on an older docs version before Next.js.

Splitting Overview and each framework adapted to different sub-article of the whole "Chat Widget Adapters" could be beneficial. I will prepare it and push on a different preview deploy hash so we will be able to compare it visually both version 👍  🙂